modules/data_release/ajax/AddPermission.php

```diff
@@ -12,25 +14,98 @@
+        $data_release_id = $_POST['data_release_id'];
+        $success         = $DB->insertIgnore(
+            'data_release_permissions',
+            array(
+             'userid'          => $userid,
+             'data_release_id' => $data_release_id,
+            )
+        );
 
-    $factory  = NDB_Factory::singleton();
-    $settings = $factory->settings();
+    } elseif (empty($_POST['data_release_id'])
+        && !empty($_POST['data_release_version'])
+    ) {
+        $userid = $_POST['userid'];
+        $data_release_version = $_POST['data_release_version'];
 
-    $baseURL = $settings->getBaseURL();
+        $IDs = $DB->pselectCol(
+            "SELECT id 
+                    FROM data_release 
+                    WHERE version=:drv",
+            array('drv' => $data_release_version)
+        );
+        foreach ($IDs as $ID) {
 
+            $success = $DB->insertIgnore(
+                'data_release_permissions',
+                array(
+                 'userid'          => $userid,
+                 'data_release_id' => $ID,
+                )
+            );
+        }
+    }
     header("Location: {$baseURL}/data_release/?addpermissionSuccess=true");
+} elseif ($_POST['action'] == 'managepermissions'
+    && $user->hasPermission('data_release_edit_file_access')
+) {
+    // 1) so with checkboxes it's not straightforward to figure out
+    //    which permissions were either set or unset during the save,
+    //    hence the TRUNCATE at the beginning
+    // 2) put it in a transaction to make sure it succeeds or not at all
+    // 3) some special characters become "_" so we wildcard match the username
+    try {
+        $DB->_PDO->beginTransaction();
+        $DB->run("TRUNCATE data_release_permissions");
+        foreach ($_POST as $key => $value) {
+            if (strpos($key, 'permissions') !== false) {
+                $parsed_user = str_replace(
+                    "_",
+                    "%",
+                    str_replace("permissions_", "", $key)
+                );
+                $userid      = $DB->pselectOne(
+                    "SELECT ID FROM users WHERE UserID LIKE :userid",
+                    array('userid' => $parsed_user)
+                );
+                foreach ($value as $k => $v) {
+                    $data_release_ids = $DB->pselect(
+                        "SELECT id FROM data_release WHERE version=:version",
+                        array('version' => $v)
+                    );
+                    foreach ($data_release_ids as $data_release_id) {
+                        $success = $DB->run(
+                            "INSERT IGNORE INTO data_release_permissions 
+                                    VALUES ($userid, {$data_release_id['id']})"
+                        );
+                    }
+                }
+            }
+        }
+        $DB->_PDO->commit();
+        header("HTTP/1.1 303 See Other");
+        //addpermissionSuccess=true/false 
```

Comment:
```suggestion
        //addpermissionSuccess=true/false
```
Removing whitespace to help Travis